packages/frontend/core/src/components/affine/quota-reached-modal/local-quota-modal.tsx

```diff
@@ -0,0 +1,55 @@
+import { ConfirmModal } from '@affine/component/ui/modal';
+import { openQuotaModalAtom } from '@affine/core/atoms';
+import { waitForCurrentWorkspaceAtom } from '@affine/core/modules/workspace';
+import { useAFFiNEI18N } from '@affine/i18n/hooks';
+import bytes from 'bytes';
+import { useAtom, useAtomValue } from 'jotai';
+import { useCallback, useEffect } from 'react';
+
+export const LocalQuotaModal = () => {
+  const t = useAFFiNEI18N();
+  const currentWorkspace = useAtomValue(waitForCurrentWorkspaceAtom);
+  const [open, setOpen] = useAtom(openQuotaModalAtom);
+
+  const checkBlobSize = useCallback((blob: Blob) => {
+    const size = BigInt(blob.size);
+    return size < bytes('100MB');
+  }, []);
+
+  const onConfirm = useCallback(() => {
+    setOpen(false);
+  }, [setOpen]);
+
+  useEffect(() => {
+    const disposable = currentWorkspace.engine.blob.onBlobSet.on(
```

Comment:
We listen to `currentWorkspace.engine.blob.isOverCapacity` here to open the model.

Reply:
In terms of design, we should limit the blob size when it is uploaded to the local. If it seems that it cannot be checked using isOverCapacity?

Reply:
Okay you can ignore this